gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/ReassureUsers.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.configcache.incremental;
+
+// CHECKSTYLE:OFF
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.gradle.api.flow.BuildWorkResult;
+import org.gradle.api.flow.FlowAction;
+import org.gradle.api.flow.FlowParameters;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.internal.cc.impl.problems.ConfigurationCacheProblems;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+// CHECKSTYLE:ON
+
+public abstract class ReassureUsers implements FlowAction<ReassureUsers.Params> {
+    private static final Logger log = LoggerFactory.getLogger(ReassureUsers.class);
+
+    interface Params extends FlowParameters {
+        @Input
+        Property<ConfigurationCacheProblems> getProblems();
```

Comment:
Can't we just inject this here instead of the `Project`? Or even in the `FlowAction` if it allows it?

Reply:
Injecting in `FlowParameters` or `FlowAction` doesn't work unfortunately :/ we have to pass it in from the plugin